src/app/features/test-records/create/views/create-test-record/create-test-record.component.ts

```diff
@@ -0,0 +1,122 @@
+import { DynamicFormService } from '@forms/services/dynamic-form.service';
+import { FormNode } from '@forms/services/dynamic-form.types';
+import { TestResultModel } from '@models/test-results/test-result.model';
+import { Actions, ofType } from '@ngrx/effects';
+import { RouterService } from '@services/router/router.service';
+import { TestRecordsService } from '@services/test-records/test-records.service';
+import { createTestResultSuccess } from '@store/test-records';
+import cloneDeep from 'lodash.clonedeep';
+import { firstValueFrom, Observable, of, Subject, take, takeUntil, tap } from 'rxjs';
+import { BaseTestRecordComponent } from '../../../components/base-test-record/base-test-record.component';
+
+@Component({
+  selector: 'app-create-test-record',
+  templateUrl: './create-test-record.component.html',
+  styleUrls: ['./create-test-record.component.scss']
+})
+export class CreateTestRecordComponent implements OnInit, OnDestroy {
+  @ViewChild(BaseTestRecordComponent) private baseTestRecordComponent?: BaseTestRecordComponent;
+
+  private destroy$ = new Subject<void>();
+
+  testResult$: Observable<TestResultModel | undefined> = of(undefined);
+  sectionTemplates$: Observable<FormNode[] | undefined> = of(undefined);
+
+  constructor(
+    private actions$: Actions,
+    private errorService: GlobalErrorService,
+    private route: ActivatedRoute,
+    private router: Router,
+    private routerService: RouterService,
+    private testRecordsService: TestRecordsService
+  ) {
+    this.router.routeReuseStrategy.shouldReuseRoute = () => false;
+  }
+
+  ngOnInit(): void {
+    this.testResult$ = this.testRecordsService.editingTestResult$.pipe(tap(editingTestResult => !editingTestResult && this.backToTechRecord()));
+
+    this.sectionTemplates$ = this.testRecordsService.sectionTemplates$;
+
+    this.routerService
+      .getQueryParam$('testType')
+      .pipe(
+        take(1),
+        tap(testType => !testType && this.backToTechRecord())
+      )
+      .subscribe(testTypeId => {
+        this.testRecordsService.contingencyTestTypeSelected(testTypeId!);
+      });
+
+    this.watchForCreateSuccess();
+  }
+
+  ngOnDestroy(): void {
+    this.testRecordsService.cancelEditingTestResult();
+    this.errorService.clearErrors();
+
+    this.destroy$.next();
+    this.destroy$.complete();
+  }
+
+  backToTechRecord(): void {
+    this.router.navigate(['../..'], { relativeTo: this.route.parent });
+  }
+
+  /**
+   * Merge all section form values into one testResult and trigger action to update testResult.
+   * @returns void
+   */
+  async handleSave(): Promise<void> {
+    const errors: GlobalError[] = [];
+    const forms = [];
+
+    if (this.baseTestRecordComponent) {
+      const { sections, defects } = this.baseTestRecordComponent;
+      if (sections) {
+        sections.forEach(section => {
+          forms.push(section.form);
+        });
+      }
+
+      if (defects) {
+        forms.push(defects.form);
+      }
+    }
```

Comment:
Any benefit to nest the `if` statements here?
```suggestion
    if (this.baseTestRecordComponent?.sections) {
      this.baseTestRecordComponent.sections.forEach(section => forms.push(section.form));
    }
    
    if (this.baseTestRecordComponent?.defects) {
      forms.push(this.baseTestRecordComponent.defects.form);
    }
```